### DISCUSSION TITLE
colorized output in the Hy REPL

### DISCUSSION THREAD

**Discussion by scauligi** [hylang/hy#2093] [Show and tell]
Here's a fun bit of code for your $HYSTARTUP file, it helps me a ton when working in the Hy REPL.
Makes it so that the REPL prints Python code in blue and Hy results in green!

```hy
(import [hy.contrib.pprint [pp pformat]])
(import [ast :as _ast])

(setv _old_unparse _ast.unparse)
(defn _new_unparse [#* args #** kwargs]
  (+ "\033[34m" (_old_unparse #* args #** kwargs) "\033[0m"))
(setv _ast.unparse _new_unparse)

(defn _pformat [#* args #** kwargs]
  (+ "\033[32m" (pformat #* args #** kwargs) "\033[0m"))

(setv repl-spy True)
(setv repl-output-fn _pformat)
```
![screenshot](https://user-images.githubusercontent.com/15003857/121739662-cdc18a00-cab0-11eb-8eaf-78b672dedade.png)

**Comment by allison-casey** [comment]
this is awesome! I ended up integrating it into my existing colorizer to add `=>` and `...` colorizing
![image](https://user-images.githubusercontent.com/18231154/121791792-e4083c80-cbbb-11eb-8fa3-df437387175c.png)

Heres a gist of my current `hyrc.hy`
https://gist.github.com/924912d86826f9b7b2e97f8bcff5e548

Most of the term color stuff was taken from ipython

**Comment by mjobuda** [comment]
How do you use this startup file / where should the hyrc.hy file be located and how initialized? I would also love to have these colors!

**Reply by scauligi** [reply]
You need to set your `$HYSTARTUP` environment variable, see https://docs.hylang.org/en/master/language/repl.html#initialization-script
(Make sure you're using either the alpha or a dev version of Hy!)
